pkg/machine/shim/host.go

```diff
@@ -354,6 +364,8 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, _ *machineDefin
 	defaultBackoff := 500 * time.Millisecond
 	maxBackoffs := 6
 
+	mc.Lock()
+	defer mc.Unlock()
```

Comment:
That looks like a deadlock. `Stop()` already has the lock.

Reply:
I see that the tests are failing because of a deadlock, but I don't think this is where? This lock call is in Start(), not in Stop(). Please let me know if I'm missing something, and you see where the deadlock is, somehow it's escaping me right now.

Reply:
I clearly lacked caffeine.  You are right!

Regarding the deadlock: check whether there are recursive functions.